python-client/giskard/core/core.py

```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional, Dict, List
+
+
+class SupportedModelTypes(Enum):
+    CLASSIFICATION = "classification"
+    REGRESSION = "regression"
+
+
+class SupportedFeatureTypes(Enum):
+    NUMERIC = "numeric"
+    CATEGORY = "category"
+    TEXT = "text"
+
+
+@dataclass
+class ModelMeta:
+    name: Optional[str]
+    model_type: SupportedModelTypes
+    feature_names: List[str]
+    classification_labels: List[str]
+    classification_threshold: float
+
+
+@dataclass
+class DatasetMeta:
+    name: Optional[str]
+    target: str
+    feature_types: Dict[str, str]
+    column_types: Dict[str, str]
```

Comment:
In fact, as it stands, if you give `column_types = feature_types` for Dataset, it will pass through:
```python
    @staticmethod
    def cast_column_to_types(df, column_types):
        current_types = df.dtypes.apply(lambda x: x.name).to_dict()
        logger.info(f"Casting dataframe columns from {current_types} to {column_types}")
        if column_types:
            try:
                df = df.astype(column_types)
            except Exception as e:
                raise ValueError("Failed to apply column types to dataset") from e
        return df
```
and would obviously crash with the following error: 
```
"Failed to apply column types to dataset"
```
was `column_types` as you wrote it here, conceived in order for the user to give low-level types @andreybavt?

Reply:
This was fixed by https://github.com/Giskard-AI/giskard/pull/652 and will be further discussed (regarding the name used for types)